update deployment modes topic

tacole02 opened the pull request:
<!--  Thanks for sending a pull request!  Before submitting:

1. Read our CONTRIBUTING.md guide
2. Rebase your PR if it gets out of sync with main
-->

#### What this PR does

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`. If changelog entry is not needed, please add the `changelog-not-needed` label to the PR.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

dimitarvdimitrov reviewed (commented):
have you thought about just deleting the docs for it? The code for read-write mode will be gone in 3.0. Are we ok with people having to switch to a 2.x version of the docs to view the read-write docs?

as for other changes - looking at the content right now I don't see any third-party deps like etcd/consul/object store, so i think we can also skip mentioning kafka

tacole02 commented:
I think you're right. If the code is gone in 3.0, and it was only ever experimental to begin with, I think we can safely delete it. 

I'm still waiting for the updated diagram from the content team. After I get that, this will be ready to merge 🥳

tacole02 commented:
Hi @dimitarvdimitrov , I added an updated diagram for ingest storage architecture in microservices mode. Based on our discussion at our last sync, I added it alongside the existing diagram for classic. Let me know if there's anything else I need to adjust here. Thanks!

56quarters commented on docs/sources/mimir/references/architecture/deployment-modes/index.md:
```diff
@@ -11,27 +11,39 @@ weight: 20
+## About monolithic mode
 
-The monolithic mode runs all required components in a single process and is the default mode of operation, which you can set by specifying `-target=all`. Monolithic mode is the simplest way to deploy Grafana Mimir and is useful if you want to get started quickly or want to work with Grafana Mimir in a development environment. To see the list of components that run when `-target` is set to `all`,  ...
+Monolithic mode runs all required components in a single process and is the default deployment mode. You enable monolithic mode by setting `-target=all`.
```
I don't really like mentioning that monolithic mode is the default here because:
* We don't want people to be running it in production but "default" makes it sound like it's the preferred way to run Mimir
* It's _not_ the default in the helm chart. It's only really the default when you run Mimir from the CLI and don't otherwise configure each component.

tacole02 replied:
Good call out, I'll remove this.

56quarters commented on docs/sources/mimir/references/architecture/deployment-modes/index.md:
```diff
@@ -41,46 +53,26 @@ Monolithic mode can be horizontally scaled out by deploying multiple Grafana Mim
+## About microservices mode
 
-In microservices mode, components are deployed in distinct processes. Scaling is per component, which allows for greater flexibility in scaling and more granular failure domains. Microservices mode is the preferred method for a production deployment, but it is also the most complex.
+Microservices mode deploys each component in separate processes, enabling independent scaling and creating granular failure domains. This deployment mode is recommended for production environments that require high availability and flexible scaling.
```
I'd rather not qualify when microservices mode is recommended for production, it's always recommended. We'd only expect people to _not_ run it in production in special circumstances that tend to be very unique to their environment or organization.

56quarters commented on docs/sources/mimir/references/architecture/deployment-modes/index.md:
```diff
@@ -11,27 +11,39 @@ weight: 20
+
+- Get started quickly with Mimir
+- Run Mimir in a development environment
+- Deploy with minimal operational complexity
```
This makes microservices mode sound like it's an optimization when it's actually the right way to be running Mimir. I would remove this and the next point. We don't want people to try running monolithic mode anywhere outside of "getting started".

56quarters reviewed (approved):
LGTM, thanks!
